ndg-commonmark/src/utils/mod.rs

```diff
@@ -125,9 +130,15 @@ pub fn extract_title_from_markdown(content: &str) -> Option<String> {
           }
         }
         // Clean the title by removing inline anchors and other NDG markup
+        let anchor_id = anchor_re.find_iter(&text).find_map(|el| {
```

Comment:
`anchor_re. find_iter(&text)` is called, but there's already a call to `anchor_re.replace_all(&text, "")` right after. If I'm not mistaken `find_map` can iterate over all matches when you only need the first one. We should use `.find(...)` directly:

```rs
let anchor_id = anchor_re.find(&text).and_then(|el| {
  el.as_str()
    .strip_prefix("{#"")?
    .strip_suffix("}")
    .map(String::from)
});
```